src/gui/widgets/main_window.py

```diff
@@ -1063,9 +1107,13 @@ def showPreviousOutput_(self):
     def setModified(self):
         if not self.modified:
             if self.gudrunFile.path:
-                self.mainWidget.setWindowModified(True)
                 self.modified = True
+                self.setWindowModified(True)
                 self.mainWidget.save.setEnabled(True)
+        if not self.proc:
+            self.timer.stop()
+            self.timer.setInterval(30000)
+            self.timer.start()
```

Comment:
```suggestion
            self.timer.start(30000)
```
This automatically stops and restarts the timer if it's already active.